Barotrauma/BarotraumaClient/ClientSource/Sprite/DeformAnimations/CustomDeformation.cs

```diff
@@ -109,9 +109,28 @@ public override float Phase
             }
         }
 
-        protected override void GetDeformation(out Vector2[,] deformation, out float multiplier, bool inverse)
+        protected override void GetDeformation(out Vector2[,] deformation, out float multiplier, bool inverse, bool isFlipped = false)
         {
-            deformation = Deformation;
+            if (isFlipped)
+            {
+                Vector2[,] flippedDeformation = new Vector2[Deformation.GetLength(0), Deformation.GetLength(1)];
```

Comment:
This causes a new array to be initialized every time the method is called. That's not good for performance: the flipped array should be cached and reused (perhaps it could be created in the constructor already).